models/migrations/v45.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import "github.com/go-xorm/xorm"
+
+func fixIndexColumnInRepoUnitTable(x *xorm.Engine) (err error) {
+	type UnitType int
+
+	// Enumerate all the unit types
+	const (
+		UnitTypeCode            UnitType = iota + 1 // 1 code
+		UnitTypeIssues                              // 2 issues
+		UnitTypePullRequests                        // 3 PRs
+		UnitTypeReleases                            // 4 Releases
+		UnitTypeWiki                                // 5 Wiki
+		UnitTypeExternalWiki                        // 6 ExternalWiki
+		UnitTypeExternalTracker                     // 7 ExternalTracker
+	)
+
+	units := map[UnitType]int{
+		UnitTypeCode:            0,
+		UnitTypeIssues:          1,
+		UnitTypeExternalTracker: 1,
+		UnitTypePullRequests:    2,
+		UnitTypeReleases:        3,
+		UnitTypeWiki:            4,
+		UnitTypeExternalWiki:    4,
+	}
```

Comment:
Why is this different than creating the RepoUnits!?
UnitTypeExternalWiki is here 4 but in when saving settings it is the Idx of models.Units, thus 6?
Definitely not correct, needs to be changed.

Reply:
@daviian I think this is correct and doesn't need any changes. `RepoUnit` contains `UnitType`, see line below.
Please see "unit.go" file. It's copied from there. The only thing I did was to replace `map[UnitType]Unit` for `map[UnitType]int` to reduce number of lines added (around 60 lines) and copied only `Unit.Idx` values. All `Index` values in `RepoUnit` table with `UnitTypeExternalWiki` should be update to `4` as it is in `units` map. Where you see 6?